kolibri/core/assets/src/core-actions.js

```diff
@@ -349,6 +443,85 @@ function stopTrackingProgress(store, Kolibri) {
   updateTimeSpent(store, Kolibri, true);
 }
 
+function saveMasteryLog(store, Kolibri) {
+  const masteryLogModel = Kolibri.resources.MasteryLog.getModel(store.core.logging.mastery.id);
+  masteryLogModel.save(_masteryLogModel(store)).then((newMasteryLog) => {
```

Comment:
@66eli77 I haven't looked at this yet, no. Want to spend some time today discussing the component structure and code layout?